halo2_debug/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+use rand_core::block::BlockRng;
+use rand_core::block::BlockRngCore;
+use rand_core::OsRng;
+use tiny_keccak::Hasher;
+
+// One number generator, that can be used as a deterministic Rng, outputing fixed values.
+pub struct OneNg {}
+
+impl BlockRngCore for OneNg {
+    type Item = u32;
+    type Results = [u32; 16];
+
+    fn generate(&mut self, results: &mut Self::Results) {
+        for elem in results.iter_mut() {
+            *elem = 1;
+        }
+    }
+}
+
+pub fn one_rng() -> BlockRng<OneNg> {
+    BlockRng::<OneNg>::new(OneNg {})
+}
+
+// Random number generator for testing
+
+pub fn test_rng() -> OsRng {
+    OsRng
+}
+
+fn keccak_hex<D: AsRef<[u8]>>(data: D) -> String {
+    let mut hash = [0u8; 32];
+    let mut hasher = tiny_keccak::Keccak::v256();
+    hasher.update(data.as_ref());
+    hasher.finalize(&mut hash);
+    hex::encode(hash)
+}
+
+// Check the a test proof against a known hash
+// Note that this function is only called in CI in "cargo test --all-fetaures"
+pub fn assert_test_proof<D: AsRef<[u8]>>(hex: &str, data: D) {
```

Comment:
I have a few comments about this method and how it's used:
1. Maybe the `D : AsRef<[u8]>` is not needed?  And we can just do `data: &[u8]`
2. What about comparing the proof itself to the hex encoding of it?  The hex string will be longer, but then we can spot the differences more easily!  Perhaps the hex string will be extremely long, in that case I agree on using a hashed version.
3. I find the name of the function strange after seeing what it does.  But when I see how it's used I understand it.  What do you think about names like `assert_data_hash_eq`? or `assert_hash_eq`?  The reason to mention `hash` in the name is because when I see the function used, the first thing that comes to my mind is that the hex string is the proof, and not a hash of the proof.